infra/blueprint-test/pkg/utils/poll.go

```diff
@@ -45,10 +55,12 @@ func Poll(t testing.TB, condition func() (bool, error), numRetries int, interval
 	}
 
 	if err != nil {
-		t.Fatalf("failed to pull provided condition after %d retries, last error: %v", numRetries, err)
+		return fmt.Errorf("failed to pull provided condition after %d retries, last error: %v", numRetries, err)
```

Comment:
nit: would custom errors be beneficial for users distinguish between errors types?

Reply:
Added.